src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -710,13 +799,26 @@ internal void EnsureChildOf(JsonTypeInfo parent)
             }
         }
 
+        internal bool TryPopulate(scoped ref ReadStack state)
+        {
+            if (EffectiveObjectCreationHandling != JsonObjectCreationHandling.Populate)
+                return false;
+
+            Debug.Assert(EffectiveConverter.CanPopulate, "Property is marked with Populate but converter cannot populate. This should have been validated in Configure");
+            Debug.Assert(state.Current.ParentObject != null, "Parent object is null");
+            Debug.Assert(!state.Current.IsPopulating, "We've called TryPopulate more than once");
+            object? value = Get!(state.Current.ParentObject);
```

Comment:
I think it might be possible to avoid boxing the property value if you moved this method to the generic `JsonPropertyInfo<T>`, at least for non-async scenaria.

Reply:
we would still need to box to populate properties though. if we move this logic to ReadJsonAndSetMember we'd need to always use continuation logic in the converters and fast-path (first `if` in the converter) can't be used